instrumentation/apache-httpclient/apache-httpclient-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v4_0/ApacheHttpClientHttpAttributesExtractor.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachehttpclient.v4_0;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.net.URI;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ApacheHttpClientHttpAttributesExtractor
+    extends HttpAttributesExtractor<HttpUriRequest, HttpResponse> {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(ApacheHttpClientHttpAttributesExtractor.class);
+
+  @Override
+  protected String method(HttpUriRequest request) {
+    return request.getMethod();
+  }
+
+  @Override
+  protected String url(HttpUriRequest request) {
+    return request.getURI().toString();
+  }
+
+  @Override
+  protected String target(HttpUriRequest request) {
+    URI uri = request.getURI();
+    String queryString = uri.getQuery();
+    return queryString != null ? uri.getPath() + "?" + queryString : uri.getPath();
```

Comment:
According to javadocs `URI#getPath()` can return nulls:

```suggestion
    String pathString = uri.getPath();
    String queryString = uri.getQuery();
    if (pathString != null && queryString != null) {
      return pathString + "?" + queryString;
    } else if (queryString != null) {
      return "?" + queryString;
    } else {
      return pathString;
    }
```